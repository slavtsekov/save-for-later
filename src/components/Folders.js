import React from 'react';

import Folder from './Folder';

const Folders = (props) => {
    const folders = ['Folder 1', 'Folder 2'];
    const handleDeleteFolder = (folderName) => {
        console.log('Removed', folderName);
    };
    return (
        <div>
            <div>
                <h3>Your Folders</h3>
            </div>
            {folders.length === 0 && <p>You have no folders at the moment.</p>}
            {
                folders.map((folder, index) => (
                    <Folder
                        key={folder}
                        text={folder}
                        onDelete={handleDeleteFolder}
                    />
                ))
            }
        </div>
    );
};

export { Folders as default };
