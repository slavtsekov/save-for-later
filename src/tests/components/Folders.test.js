import React from 'react';
import { shallow } from 'enzyme';
import Folders from '../../components/Folders';

test('should render Folders correctly', () => {
    const wrapper = shallow(<Folders />);
    expect(wrapper).toMatchSnapshot();
});
