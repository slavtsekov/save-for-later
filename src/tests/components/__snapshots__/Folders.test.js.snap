// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Folders correctly 1`] = `
<div>
  <div>
    <h3>
      Your Folders
    </h3>
  </div>
  <Folder
    key="Folder 1"
    onDelete={[Function]}
    text="Folder 1"
  />
  <Folder
    key="Folder 2"
    onDelete={[Function]}
    text="Folder 2"
  />
</div>
`;
